core/src/test/java/apoc/merge/MergeTest.java

```diff
@@ -136,23 +124,59 @@ public void testMergeWithEmptyIdentityPropertiesShouldFail() {
             }
         }
     }
-    
+
+    @Test
+    public void testMergeNodeWithNullLabelsShouldFail() {
+        try {
+            testCall(db, "CALL apoc.merge.node([null, null], {name:'John'}) YIELD node RETURN node",
+                    row -> assertTrue(row.get("node") instanceof Node));
+            fail();
+        } catch (QueryExecutionException e) {
+            assertEquals(e.getMessage(), "Failed to invoke procedure `apoc.merge.node`: Caused by: java.lang.IllegalArgumentException: " +
+                    "The list of label names may not contain any null or empty String values. If you wish to merge a node without a label, pass an empty list instead.");
+        }
+    }
+
+    @Test
+    public void testMergeNodeWithEmptyLabelListShouldFail() {
```

Comment:
It would be good to also have a test where not all labels are empty, like CALL apoc.merge.node(['Person' , ''], {name:'John'})...